# Модуль 3. Практическая работа 

## Цели практической работы
1. Закрепить полученные знания о дискретных случайных величинах.
2. Ближе познакомиться с характеристиками случайных величин, изучить на практике дискретные распределения: Бернулли и Пуассона.


## Что входит в практическую работу
  3 практические задачи

## Что оценивается
 
*  Умение вычислять характеристики случайных величин на основе табличного или графического представления дискретных случайных величин и с использованием формул. \
*  Умение работать с формулами распределений Пуассона и Бернулли.


## Как отправить задание на проверку 

Выполните задания, сохраните изменения при помощи опции Save and Checkpoint на вкладке меню File или кнопки Save and Checkpoint на панели инструментов. Отправьте итоговый файл Jupyter Notebook (в формате .ipynb) или ссылку на него через форму ниже.

## Задача 1 

Имеется партия из десяти деталей. Вероятность, что деталь оказалась бракованной, для данной партии составляет $4/5 = 0.8.$ Определите, с какой вероятностью половина деталей окажется бракованной.

Напишите функцию для расчёта этой вероятности. Используете биномиальное распределение.

 




In [1]:
# Ваше решение

import math

def probability_half_defective_parts(n, p):
    k = n // 2  # количество бракованных деталей для половины
    probability = math.comb(n, k) * (p ** k) * ((1 - p) ** (n - k))
    return probability

# Пример использования функции для заданной вероятности p = 0.8 и количества деталей n = 10
p = 0.8
n = 10
result = probability_half_defective_parts(n, p)
print("Вероятность того, что половина деталей окажется бракованной:", result)

Вероятность того, что половина деталей окажется бракованной: 0.026424115199999976


### Задача 2.  

Вратарь последовательно отражает семь ударов в ворота от игрока. Вероятность этого составляет $0.3$ каждый раз независимо от характеристик удара. Найдите вероятность, с которой вратарь отразит хотя бы три удара. Используете биномиальное распределение.





In [2]:
# Ваше решение

def probability_at_least_three_deflections(n, p):
    probability = 0
    for k in range(3, n + 1):
        probability += math.comb(n, k) * (p ** k) * ((1 - p) ** (n - k))
    return probability

# Пример использования функции для заданной вероятности p = 0.3 и количества ударов n = 7
p = 0.3
n = 7
result = probability_at_least_three_deflections(n, p)
print("Вероятность отражения хотя бы трех ударов вратарем:", result)

Вероятность отражения хотя бы трех ударов вратарем: 0.3529304999999999


### Задача 3.  

Представьте, что вы сделали бросок пары игральных кубиков десять раз подряд. При каждом броске получаете два значения на двух кубиках. Рассмотрите вероятность, при которой один раз выпадет пара из двух шестёрок. Найдите её:
с помощью биномиального распределения;
с помощью распределения Пуассона, используя закон редких событий. 


In [3]:
# Ваше решение

from scipy.stats import binom, poisson

# Количество бросков
n = 10

# Вероятность выпадения шестёрки на одном кубике
p = 1/6

# Вероятность выпадения пары шестёрок при одном броске
p_pair = p * p

# Вероятность выпадения пары шестёрок хотя бы один раз за 10 бросков, используя биномиальное распределение
prob_binomial = 1 - binom.cdf(0, n, p_pair)

# Вероятность выпадения пары шестёрок, используя распределение Пуассона (при условии, что количество бросков большое и вероятность мала)
lambda_ = n * p_pair
prob_poisson = poisson.pmf(1, lambda_)

print("Вероятность выпадения пары шестёрок хотя бы один раз за 10 бросков (биномиальное распределение):", prob_binomial)
print("Вероятность выпадения пары шестёрок (распределение Пуассона):", prob_poisson)


Вероятность выпадения пары шестёрок хотя бы один раз за 10 бросков (биномиальное распределение): 0.24550661610055657
Вероятность выпадения пары шестёрок (распределение Пуассона): 0.21040698011026843


Из результатов, полученных с использованием биномиального распределения и распределения Пуассона, видно, что оба метода дают близкие значения вероятности выпадения пары шестёрок хотя бы один раз за 10 бросков. Однако вероятность, рассчитанная с использованием биномиального распределения (0.2455), немного выше, чем вероятность, рассчитанная с использованием распределения Пуассона (0.2104).

Это может быть связано с тем, что распределение Пуассона является лишь приближением биномиального распределения при большом количестве попыток и малой вероятности успеха. В данном случае у нас 10 бросков, что не является очень большим числом, поэтому биномиальное распределение может быть более точным для данной задачи.

Тем не менее, оба метода дали сопоставимые результаты, и вывод заключается в том, что есть некоторая вероятность того, что пара шестёрок выпадет хотя бы один раз за 10 бросков, но эта вероятность не очень высока.